impl/maven-impl/src/main/java/org/apache/maven/impl/model/DefaultModelInterpolator.java

```diff
@@ -101,21 +104,14 @@ interface InnerInterpolator {
     @Override
     public Model interpolateModel(
             Model model, Path projectDir, ModelBuilderRequest request, ModelProblemCollector problems) {
-        InnerInterpolator innerInterpolator = createInterpolator(model, projectDir, request, problems);
-        return new MavenTransformer(innerInterpolator::interpolate).visit(model);
+        return new MavenTransformer(createInterpolator(model, projectDir, request, problems)::interpolate).visit(model);
     }
 
     private InnerInterpolator createInterpolator(
             Model model, Path projectDir, ModelBuilderRequest request, ModelProblemCollector problems) {
-
-        Map<String, Optional<String>> cache = new HashMap<>();
```

Comment:
what cache, its only **used one time**? there is no reference there is no cache, every one has its own.
- https://github.com/pmd/pmd/issues/5770